bin/specextract

```diff
@@ -1389,21 +1409,91 @@ def spectra(args):
 
                 v1(f"{phafile} AREASCAL keyword set to the 'blanksky' BKGSCAL{ccdid} value for automated background spectrum scaling while spectral modeling.  Set AREASCAL to 1.0 if 'blanksky' spectrum is being used for background subtraction.")
 
-    try:
-        outreg.close()
-    except NameError:
-        outreg = None
-                
     try:
         return pardict
     except NameError:
         return None
 
-    
 
-def resps(args):
+
+def _resps_preprocess_wrapper(args_dict):
+    key_id = args_dict["key_id"]
+
+    filename = args_dict["filename"]
+    fullfile = args_dict["fullfile"]
+    full_outroot = args_dict["full_outroot"]
+
+    instrument = args_dict["instrument"]
+    chip_id = args_dict["chip_id"]
+
+    dobpix = args_dict["dobpix"]
+    dobkgresp = args_dict["dobkgresp"]
+
+    tmpdir = args_dict["tmpdir"]
+    clobber = args_dict["clobber"]
+    verbose = args_dict["verbose"]
+
+    try:
+        asol = args_dict["asol"]
+        asolstat = True
+        asphist_name = args_dict["asphist_name"]
+    except KeyError:
+        asphist = args_dict["asphist"]
+        asolstat = False
+
+    if dobpix:
+        try:
+            bpix = args_dict["bpix"]
+        except KeyError:
+            bpix = None
+
+    try:
+        dtf = args_dict["dtf"]
+    except KeyError:
+        dtf = None 
+
+    kdict = {}
+
+    with new_pfiles_environment(ardlib=True,copyuser=False):
+        if dobpix:
+            set_badpix(filename,bpix,instrument,verbose)
+
+        if asolstat:
+            #########################################
+            #
+            # create aspect histograms if necessary
+            # to pass along toe mkarf
+            #
+            #########################################
+
+            with spec.suppress_stdout_stderr():
+                try:
+                    asphist = mk_asphist(asol,asphist_name,fullfile,full_outroot,
+                                         dtf,instrument,chip_id,
+                                         verbose,clobber,tmpdir)
+                    
+                except OSError as exc:
+                    if dobkgresp:
+                        raise IOError(f"Failed to create aspect histogram file for {fullfile}") from exc
+
+                    asphist = None
+
+        kdict["asphist"] = asphist
+
+        if instrument == "ACIS":
+            rmftool = args_dict["rmftool"]
+                        
+            kdict["tdetwmap"], kdict["feffile"] = resp_setup(args_dict,asphist,rmftool)
+
+        return {key_id : kdict}
```

Comment:
```suggestion
    return {key_id : kdict}
```

Techincally we don't have to do this, but it makes it a bit clearer when the return happens.